common/achievement/bucket.ts

```diff
@@ -55,19 +55,23 @@ export const bucketCreatorDefs: BucketCreatorDefs = {
             const subcourseBuckets = context.subcourse
                 .map((subcourse) => createLectureBuckets(subcourse, LectureFilter.start))
                 .reduce((acc, val) => acc.concat(val), []);
-            return { bucketKind: 'time', buckets: [...matchBuckets, ...subcourseBuckets] };
+            const buckets = [...matchBuckets, ...subcourseBuckets].sort((a, b) => a.startTime.getTime() - b.startTime.getTime());
+            return { bucketKind: 'time', buckets: buckets };
         },
     },
     by_lecture_participation: {
         function: (bucketContext): GenericBucketConfig<TimeBucket> => {
             const { context } = bucketContext;
             const matchBuckets = context.match
                 .map((match) => createLectureBuckets(match, LectureFilter.participation))
-                .reduce((acc, val) => acc.concat(val), []);
+                .reduce((acc, val) => acc.concat(val), [])
+                .filter((bucket) => bucket.startTime.getTime() <= new Date().getTime() + 30 * 60000);
```

Comment:
We should avoid magic numbers. Please move it to a constant with a good name and some description.